sql/src/main/java/io/crate/analyze/CopyStatementAnalyzer.java

```diff
@@ -95,20 +96,36 @@ public CopyToAnalyzedStatement convertCopyTo(CopyTo node, Analysis analysis) {
         Context context = new Context(analysisMetaData, analysis.parameterContext(), tableRelation, false);
         Settings settings = processGenericProperties(node.genericProperties(), context);
         Symbol uri = context.processExpression(node.targetUri());
+        WhereClause whereClause = null;
+        if (node.whereClause().isPresent()) {
+            WhereClauseAnalyzer whereClauseAnalyzer = new WhereClauseAnalyzer(analysisMetaData, tableRelation);
+            whereClause = whereClauseAnalyzer.analyze(
+                    context.expressionAnalyzer.generateWhereClause(node.whereClause(), context.expressionAnalysisContext));
+        }
 
         List<Symbol> outputs = new ArrayList<>();
         QuerySpec querySpec = new QuerySpec();
-        PartitionName partition = null;
+        List<String> partitions = ImmutableList.of();
         if (!node.table().partitionProperties().isEmpty()) {
-            partition = PartitionPropertiesAnalyzer.toPartitionName(
+            PartitionName partitionName = PartitionPropertiesAnalyzer.toPartitionName(
                     tableRelation.tableInfo(),
                     node.table().partitionProperties(),
                     analysis.parameterContext().parameters());
 
-            if (!partitionExists(tableRelation.tableInfo(), partition)) {
-                throw new PartitionUnknownException(tableRelation.tableInfo().ident().fqn(), partition.ident());
+            if (!partitionExists(tableRelation.tableInfo(), partitionName)) {
+                throw new PartitionUnknownException(tableRelation.tableInfo().ident().fqn(), partitionName.ident());
+            }
+            partitions = ImmutableList.of(partitionName.asIndexName());
+        }
+
+        if (whereClause == null) {
+            querySpec.where(new WhereClause(null, null, partitions));
+        } else {
+            if (!whereClause.partitions().equals(partitions)) {
+                throw new IllegalArgumentException("Given partition ident does not match partition evaluated from where clause");
```

Comment:
does this also raise an error if  I do COPY users PARTITION (p=1) where gender = 'female') ? I think that would be a valid use case